packages/eslint-plugin/src/rules/no-misused-promises.ts

```diff
@@ -137,6 +158,164 @@ export default util.createRule<Options, 'conditional' | 'voidReturn'>({
       }
     }
 
+    function checkAssignments(node: TSESTree.AssignmentExpression): void {
```

Comment:
Naming nit on this and the other new checkers except `checkVariableDeclaration`, they're singular not plural since they only handle one node at a time:

```suggestion
    function checkAssignment(node: TSESTree.AssignmentExpression): void {
```